File: WebPresenter/PresentationFundamentals.cs
using System;
using WebPresenter.Models;

namespace WebPresenter {
    [Serializable]
    public class PresentationFundamentals : ICloneable {
        public string Name { get; set; }
        public string OwnerName { get; set; }
        public string Title { get; set; }
        
        public PresentationFundamentals() {}

        public PresentationFundamentals(PresentationData data) {
            Name = data.Name;
            OwnerName = data.OwnerName;
            Title = data.Title;
        }

        public override string ToString() {
            return $"{nameof(Name)}: {Name}, {nameof(OwnerName)}: {OwnerName}, {nameof(Title)}: {Title}";
        }

        public object Clone() {
            return new PresentationFundamentals {
                Name = Name,
                OwnerName = OwnerName,
                Title = Title
            };
        }
    }
}

File: WebPresenter/ClientApp/src/app/types/presentationFundamentals.ts
import {HttpClient} from "@angular/common/http";

export class PresentationFundamentals {
  public name: string;
  public ownerName: string;
  public title: string;

  constructor(name: string, ownerName: string, title = "") {
    this.name = name;
    this.ownerName = ownerName;
    this.title = title;
  }

  public createInDB(http: HttpClient, callback: (successful: boolean, error: Error) => void) {
    http.post("/data/presentationData/", this)
      .subscribe(
        () => callback(true, null),
        error => callback(false, error)
      );
  }

  public start(http: HttpClient, callback: (successful: boolean, id: any, error: Error) => void) {
    http.post("/data/presentations/", this)
      .subscribe(
        // @ts-ignore
        response => callback(true, response.content, undefined),
        error => callback(false, undefined, error)
      );
  }

  public deleteInDB(http: HttpClient, callback: (successful: boolean, error: Error) => void) {
    http.delete(`/data/presentationData/${this.ownerName}/${this.name}`)
      .subscribe(
        () => callback(true, null),
        error => callback(false, error)
      );
  }
}
